podcasts/Sharing/SharingModal.swift

```diff
@@ -206,20 +206,46 @@ extension SharingModal.Option {
 
         progress.wrappedValue = 0.01
 
-        let fileURL = try await destination.export(info: imageInfo,
-                                        style: style,
-                                        episode: episode,
-                                        startTime: CMTime(seconds: clipTime.start, preferredTimescale: 600),
-                                        duration: CMTime(seconds: clipTime.end - clipTime.start, preferredTimescale: 600),
-                                        progress: nsProgress
-        )
+        let url = FileManager.default.temporaryDirectory.appendingPathComponent("video_export-\(clipUUID)-\(style.hashValue)-\(clipTime.start)-\(clipTime.end)-\(destination.hashValue)", conformingTo: .mpeg4Movie)
+        let fileURL: URL
+        if FileManager.default.fileExistsAtURL(url) {
+            fileURL = url
+        } else {
+            fileURL = try await destination.export(info: imageInfo,
+                                            style: style,
+                                            episode: episode,
+                                            startTime: CMTime(seconds: clipTime.start, preferredTimescale: 600),
+                                            duration: CMTime(seconds: clipTime.end - clipTime.start, preferredTimescale: 600),
+                                            progress: nsProgress,
+                                            to: url
+            )
+        }
 
         progress.wrappedValue = nil
 
         if destination == .instagram {
             return try? Data(contentsOf: fileURL) // For some reason, I couldn't get this to work with just a URL
         } else {
-            return fileURL as NSURL // Third party apps need URLs and won't accept Data
+            func format(time: TimeInterval) -> String {
```

Comment:
Nit: Can this one be a new extension for TimeInterval? Like the one Sentry has.

Reply:
This was really just intended to not repeat the code in this function. I've added it as a `fileprivate` extension here so it's a bit cleaner.